src/mpi/request/testall.c

```diff
@@ -35,148 +35,26 @@ int MPI_Testall(int count, MPI_Request array_of_requests[], int *flag,
 #define FUNCNAME MPIR_Testall_impl
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-int MPIR_Testall_impl(int count, MPI_Request array_of_requests[], int *flag,
+int MPIR_Testall_impl(int count, MPIR_Request *request_ptrs[], int *flag,
```

Comment:
@pavanbalaji @hajimefu Do we need `flag` here? This code does not use it if I read correctly. Is it just to comply with other `MPIR_*_impl` series having the same header as `MPI_*` except `MPIR_Request` is used ?

Reply:
I think you're right. We can remove it. Apparently we blindly copied the existing function signature and did not carefully look at each parameter.

Reply:
Actually, the better question is whether the associated device call is going to set the flag (is the "device" layer supposed to set the user flag ?) since this is going to be turn into `MPID_Testall`, if yes then we have to keep it.